h2/src/test/org/h2/test/db/TestOutOfMemory.java

```diff
@@ -152,68 +154,80 @@ private void testUpdateWhenNearlyOutOfMemory() throws Exception {
         if (config.memory) {
             return;
         }
-        deleteDb("outOfMemory");
-        String url = getURL("outOfMemory", true);
+        deleteDb(DB_NAME);
+
+        ProcessBuilder processBuilder = buildChild(
+                DB_NAME + ";MAX_OPERATION_MEMORY=1000000",
+                MyChild.class,
+                "-XX:+UseParallelGC",
+//                "-XX:+UseG1GC",
+                "-Xmx128m");
 //*
-        String selfDestructor = SelfDestructor.getPropertyString(1);
-        String args[] = {System.getProperty("java.home") + File.separatorChar + "bin" + File.separator + "java",
-                        "-Xmx128m", "-XX:+UseParallelGC", // "-XX:+UseG1GC",
-                        "-cp", getClassPath(),
-                        selfDestructor,
-                        Child.class.getName(), url};
-        ProcessBuilder pb = new ProcessBuilder()
-                            .redirectOutput(ProcessBuilder.Redirect.INHERIT)
-                            .redirectError(ProcessBuilder.Redirect.INHERIT)
-                            .command(args);
-        Process p = pb.start();
-        int exitValue = p.waitFor();
-        assertEquals("Exit code", 0, exitValue);
+        processBuilder.start().waitFor();
 /*/
-        Child.main(url);
+        List<String> args = processBuilder.command();
+        for (Iterator<String> iter = args.iterator(); iter.hasNext(); ) {
+            String arg = iter.next();
+            if(arg.equals(MyChild.class.getName())) {
+                iter.remove();
+                break;
+            }
+            iter.remove();
+        }
+        MyChild.main(args.toArray(new String[0]));
 //*/
```

Comment:
leftover commented out code?

Reply:
I left it intentionally, so in case something goes wrong, by removing a single slash on "//*" line, you can flip it to run in-process for easy debugging.